File: src/com/crazicrafter1/lootcrates/listeners/BaseListener.java
package com.crazicrafter1.lootcrates.listeners;

import com.crazicrafter1.lootcrates.Main;
import org.bukkit.Bukkit;
import org.bukkit.event.Listener;

public class BaseListener implements Listener {

    protected static Main plugin = Main.get();

    public BaseListener(Main plugin) {
        Bukkit.getPluginManager().registerEvents(this, plugin);
    }

}

File: src/main/java/com/crazicrafter1/lootcrates/listeners/BaseListener.kt
package com.crazicrafter1.lootcrates.listeners

import com.crazicrafter1.lootcrates.LCMain
import org.bukkit.Bukkit
import org.bukkit.event.Listener

open class BaseListener(plugin: LCMain?) : Listener {
    init {
        Bukkit.getPluginManager().registerEvents(this, plugin!!)
    }

    companion object {
        protected var plugin: LCMain? = LCMain.Companion.get()
    }
}
